### PR TITLE
Walker refactor

### LINKED ISSUE #92: Undo inheritance of Walker2D from Hopper
Make Walker2D its own class that doesn't inherit from Hopper - duplicate code is OK as each env serves as an independent reference.

### PR DESCRIPTION
**Pull request by sgillen**
Closes #92. 

I verified that the "new" walker performs the same as before the refactor. I also have a set of parameters that work consistently for walker+ppo, since I saw them missing in the demo colab (I was not able to find good parameters for hopper).

Using:
```python
 'walker2d': functools.partial(
      ppo.train, num_timesteps = 100_000_000, log_frequency = 10,
      reward_scaling = 1, episode_length = 1000, normalize_observations = True,
      action_repeat = 1, unroll_length = 20, num_minibatches = 512,
      num_update_epochs = 20, discounting = 0.99, learning_rate = 5e-5,
      entropy_cost = 0.0005, num_envs = 2048, batch_size = 32
  ),
```

I get the attached reward curve / gif. I can also include changes to the training notebook in this PR, with the minor changes needed to include the walker. Not sure what if any policy there is for notebooks in PRs since they can be painful to VC, and also not sure how to update the colab version. 

brax_walker_reward
brax_walker2

### CODE REVIEW

**Comment by KarolisRam** [comment]
About the provided hyper-parameters:
TLDR: I couldn't reproduce the results after retraining for 30+ times, so I did some sweeping to come up with HPs that work for me and train 5-10x faster.

Details:
Maybe I was training it wrong, but the HPs in Colab for other envs work fine for me. Or perhaps it's due to changes in [friction](https://github.com/google/brax/commit/e782786b77b487d1a648e81c26b6ae075b3d378c)?
The HPs provided above take 38mins to train on my single GPU (2080 ti) machine and give mean reward of only 2139 with the following distribution (32 retraining runs, final reward on the learning curve):
PR-HPs

When doing my sweeps I tried to stay close to HP ranges that were used in other envs, especially the `batch_size` and `num_minibatches`, which impacts the training speed. I came up with two sets of HPs: one trains in 3min and is slightly worse, while the better one takes 6min to train. I retrained these two 60+ times each to come up with the distributions below:

3 min one, mean reward of 4253:
```
 'walker2d': functools.partial(
      ppo.train, num_timesteps = 50_000_000, log_frequency = 10,
      reward_scaling = 0.1, episode_length = 1000, normalize_observations = True,
      action_repeat = 1, unroll_length = 20, num_minibatches = 32,
      num_update_epochs = 8, discounting = 0.97, learning_rate = 0.0003,
      entropy_cost = 0.001, num_envs = 2048, batch_size = 256
  ),
```
noble-sweep-43

6 min one, mean reward of 4844:
```
 'walker2d': functools.partial(
      ppo.train, num_timesteps = 100_000_000, log_frequency = 10,
      reward_scaling = 1, episode_length = 1000, normalize_observations = True,
      action_repeat = 1, unroll_length = 20, num_minibatches = 32,
      num_update_epochs = 8, discounting = 0.97, learning_rate = 0.0003,
      entropy_cost = 0.01, num_envs = 2048, batch_size = 512
  ),
```
sunny-sweep-134

**Comment by erikfrey** [comment]
@KarolisRam thank you for posting these hparams!  I've added them to the colab in https://github.com/google/brax/commit/7229ee910aaf6a96a30cf55f50fe345feaa31bc0 for everyone else to enjoy.   :-)
